strawberry/extensions/parser_cache.py

```diff
@@ -0,0 +1,22 @@
+from functools import lru_cache
+
+from graphql.language import DocumentNode
+
+from strawberry.extensions.base_extension import Extension
+from strawberry.schema.execute import parse_document
+
+
+def UseParserCache(maxsize: int = None):
+    @lru_cache(maxsize=maxsize)
+    def cached_parse_document(query: str) -> DocumentNode:
+        return parse_document(query)
+
+    class _UseParserCache(Extension):
+        def on_parsing_start(self):
+            execution_context = self.execution_context
+
+            execution_context.graphql_document = cached_parse_document(
+                execution_context.query,
+            )
+
+    return _UseParserCache
```

Comment:
we should probably add docstrings for these, it would be useful to know when I should use it and what maxsize I should be setting :)

Reply:
Yep good idea. Also I wanted to make a docs page that has all the extensions on it. Maybe one page per extension. What do you think?

Reply:
Yes, do you think it is worth extracting the extensions from the source code? so we don't have to duplicate docs? I can work on that

Reply:
Do you mean extracting the docs from the source code? That would be cool but I also don't see a problem with creating doc pages for each extension as well. As long as we have a template we can work off it should be fine.

Reply:
ok, let's do it manually for now :)

I think it would be nice to get API docs generated from code, but we can do that in future!